arch/arm/soc/atmel_sam/common/arm_mpu_regions.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <arch/arm/cortex_m/mpu/arm_mpu.h>
+
+#include "arm_mpu_mem_cfg.h"
+
+/* System Private Peripheral Bus */
+#define PPB_BASE  0xE0000000
+
+static struct arm_mpu_region mpu_regions[] = {
```

Comment:
I suspect we can drastically reduce the size of this.
Right now there is a background mapping set in the MPU which grants supervisor mode RW (no execute) access to the entire memory map, which acts as a least-precedence region for the entire address space.
We will need to keep the FLASH_0 region.
But with the background map I think we can discard RAM_0, PERIPHERAL_0, and PPB_0. Hence freeing up more regions for use with memory domains.
@agross-linaro @chunlinhan is my thinking sound on this?